etc/httpd/global.conf

```diff
@@ -884,3 +884,6 @@ RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?products(/?)$ /$1firefox/products/ [L,R=302]
 # bug 1133004
 RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?firefox/products(/?)$ /b/$1firefox/products$2 [PT]
 
+# bug 1123399
+RewriteRule ^/(\w{2,3}(?:-\w{2})?/)?thunderbird/all(\.html)?$ /$1thunderbird/all/ [L,R=301]
```

Comment:
I think you can make the `.html` non-optional in this one. `thunderbird/all` will be caught by the next line.